cocotb/_build_libs.py

```diff
@@ -0,0 +1,440 @@
+            )
+            icarus_compile = False
+        else:
+            icarus_path = os.path.dirname(os.path.dirname(iverilog_path))
+            icarus_extra_lib = ["vpi"]
+            icarus_extra_lib_path = [os.path.join(icarus_path, "lib")]
+
+    if icarus_compile:
+        ext += _get_common_lib_ext(include_dir, share_lib_dir, sim_define="ICARUS")
+        icarus_vpi_ext = _get_vpi_lib_ext(
+            include_dir=include_dir,
+            share_lib_dir=share_lib_dir,
+            sim_define="ICARUS",
+            extra_lib=icarus_extra_lib,
+            extra_lib_dir=icarus_extra_lib_path,
+        )
+
+        ext.append(icarus_vpi_ext)
+
+    #
+    #  Modelsim/Questa
+    #
+    vsim_path = find_executable("vdbg")
+    modelsim_compile = True
+    modelsim_extra_lib = []
+    modelsim_extra_lib_path = []
+    logger.info("Compiling libraries for Modelsim/Questa")
+    if os.name == "nt":
+        if vsim_path is None:
+            logger.warning(
+                "Modelsim/Questa executable (vdbg) not found. No VPI interface will be available."
+            )
+            modelsim_compile = False
+        else:
+            modelsim_bin_dir = os.path.dirname(vsim_path)
+            modelsim_extra_lib = ["mtipli"]
+            modelsim_extra_lib_path = [modelsim_bin_dir]
+
+    if modelsim_compile:
+        ext += _get_common_lib_ext(include_dir, share_lib_dir, sim_define="MODELSIM")
+        modelsim_vpi_ext = _get_vpi_lib_ext(
+            include_dir=include_dir,
+            share_lib_dir=share_lib_dir,
+            sim_define="MODELSIM",
+            extra_lib=modelsim_extra_lib,
+            extra_lib_dir=modelsim_extra_lib_path,
+        )
+
+        ext.append(modelsim_vpi_ext)
+
+    if vsim_path is None:
+        logger.warning(
+            "Modelsim/Questa executable (vdbg) executable not found. No FLI interface will be available."
+        )
+    else:
+        modelsim_dir = os.path.dirname(os.path.dirname(vsim_path))
+        modelsim_include_dir = os.path.join(modelsim_dir, "include")
+        if os.path.isfile(os.path.join(modelsim_include_dir, "mti.h")):
+            fli_ext = Extension(
+                os.path.join("cocotb", "libs", "modelsim", "libfli"),
+                include_dirs=[include_dir, modelsim_include_dir],
+                libraries=["gpi", "gpilog", "stdc++"] + modelsim_extra_lib,
+                library_dirs=modelsim_extra_lib_path,
+                sources=[
+                    os.path.join(share_lib_dir, "fli", "FliImpl.cpp"),
+                    os.path.join(share_lib_dir, "fli", "FliCbHdl.cpp"),
+                    os.path.join(share_lib_dir, "fli", "FliObjHdl.cpp"),
+                ],
+                extra_link_args=["-Wl,-rpath,$ORIGIN"],
+            )
+
+            ext.append(fli_ext)
+
+        else:
+            logger.warning(
+                "Building FLI interface for Modelsim failed! No mti.h available."
```

Comment:
```suggestion
                "Cannot build FLI interface for Modelsim/Questa - no mti.h available."
```
is closer to what is actually happening.

Reply:
Or perhaps more clearly, which would have probably helped you debug:
```suggestion
                "Building FLI interface for Modelsim failed! "
                "The mti.h header for '{}' was not found at '{}' "
                .format(vsim_path, mti_path)
```